pkg/controller/master/backup/backupcontent.go

```diff
@@ -340,14 +349,37 @@ func (h *ContentHandler) reconcileLonghornVolumes(vm *kubevirtv1.VirtualMachine)
 			return fmt.Errorf("failed to get volume %s/%s, error: %s", name, vm.Namespace, err.Error())
 		}
 
-		volCpy := volume.DeepCopy()
 		if volume.Status.State == types.VolumeStateDetached || volume.Status.State == types.VolumeStateDetaching {
-			volCpy.Spec.NodeID = volume.Status.OwnerID
-		}
+			// Volume is already propose to a node. No need to update.
+			if volume.Spec.NodeID == volume.Status.OwnerID {
+				return nil
+			}
+
+			proposedNodeID := volume.Status.OwnerID
+
+			// HACK: Mind any further upgrade that breaks `spec.nodeID`
+			//
+			// The reason choosing JSON patch[1] is that for current Longhorn
+			// `VolumeSpec`, there is no field get a `omitempty` tag, so field
+			// like `Spec.Size` will serialize to 0. Using JSON-merge patch[2]
+			// may lead to apply unexpected default value. Also, CRD does not
+			// support strategy patch[3].
+			//
+			// [1]: https://datatracker.ietf.org/doc/html/rfc6902
+			// [2]: https://datatracker.ietf.org/doc/html/rfc7386
+			// [3]: https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#advanced-features-and-flexibility
```

Comment:
in what case the `spec.migratable` is equal to false, based on my testing this value always exists and defaults to `true`. 
Additionally, can we check with the LH team to check why this field is missing and we may set it to default to true if this field is nil, and then we can call the update method directly?

Reply:
Here is the payload of our HTTP PUT payload  generated by `Update` method.

<details>
<summary>expand</summary>

```json
{
  "kind": "Volume",
  "apiVersion": "longhorn.io/v1beta1",
  "metadata": {
    "name": "pvc-397acf57-d073-41c7-938a-e5c56f186aa3",
    "namespace": "longhorn-system",
    "uid": "59c602c4-2b4f-472d-a350-71ea312157aa",
    "resourceVersion": "79519",
    "generation": 3,
    "creationTimestamp": "2021-06-24T08:51:53Z",
    "labels": {
      "longhornvolume": "pvc-397acf57-d073-41c7-938a-e5c56f186aa3"
    },
    "finalizers": [
      "longhorn.io"
    ],
    "managedFields": [
      {
        "manager": "longhorn-manager",
        "operation": "Update",
        "apiVersion": "longhorn.io/v1beta1",
        "time": "2021-06-24T08:51:53Z",
        "fieldsType": "FieldsV1",
        "fieldsV1": {
          "f:metadata": {
            "f:finalizers": {
              ".": {},
              "v:\"longhorn.io\"": {}
            },
            "f:labels": {
              ".": {},
              "f:longhornvolume": {}
            }
          },
          "f:spec": {},
          "f:status": {}
        }
      }
    ]
  },
  "spec": {
    "size": "10737418240",
    "frontend": "blockdev",
    "fromBackup": "",
    "numberOfReplicas": 3,
    "dataLocality": "disabled",
    "staleReplicaTimeout": 30,
    "nodeID": "h0",
    "engineImage": "longhornio/longhorn-engine:v1.1.1",
    "recurringJobs": null,
    "baseImage": "",
    "Standby": false,
    "diskSelector": null,
    "nodeSelector": null,
    "disableFrontend": false,
    "revisionCounterDisabled": false,
    "lastAttachedBy": "",
    "accessMode": "rwx"
  },
  "status": {
    "ownerID": "h0",
    "state": "detached",
    "robustness": "unknown",
    "currentNodeID": "",
    "currentImage": "longhornio/longhorn-engine:v1.1.1",
    "kubernetesStatus": {
      "pvName": "pvc-397acf57-d073-41c7-938a-e5c56f186aa3",
      "pvStatus": "Bound",
      "namespace": "default",
      "pvcName": "bb-1-rootdisk-mklow",
      "lastPVCRefAt": "",
      "workloadsStatus": [
        {
          "podName": "virt-launcher-bb-1-vjmsd",
          "podStatus": "Running",
          "workloadName": "bb-1",
          "workloadType": "VirtualMachineInstance"
        }
      ],
      "lastPodRefAt": "2021-06-24T08:59:40Z"
    },
    "conditions": {
      "restore": {
        "type": "restore",
        "status": "False",
        "lastProbeTime": "",
        "lastTransitionTime": "2021-06-24T08:52:02Z",
        "reason": "",
        "message": ""
      },
      "scheduled": {
        "type": "scheduled",
        "status": "True",
        "lastProbeTime": "",
        "lastTransitionTime": "2021-06-24T09:00:30Z",
        "reason": "",
        "message": ""
      },
      "toomanysnapshots": {
        "type": "toomanysnapshots",
        "status": "False",
        "lastProbeTime": "",
        "lastTransitionTime": "2021-06-24T08:52:02Z",
        "reason": "",
        "message": ""
      }
    },
    "lastBackup": "",
    "lastBackupAt": "",
    "pendingNodeID": "",
    "frontendDisabled": false,
    "restoreRequired": false,
    "restoreInitiated": false,
    "remountRequestedAt": "",
    "expansionRequired": false,
    "isStandby": false,
    "actualSize": 0,
    "lastDegradedAt": "",
    "shareEndpoint": "",
    "shareState": ""
  }
}
```
</details>

I think the root cause is [Longhorn v1.1.0 lacks Migratable field][1], so a PUT (replace) operation will lead `Migratable` to set to false by default.

[1]: https://github.com/longhorn/longhorn-manager/blob/v1.1.0/types/resource.go#L76-L94

Reply:
Wait that's weird. We're using 1.1.0 why we need the Migratable field.

Reply:
Maybe we just need an update to v1.1.1, since our helm chart already uses Longhorn v1.1.1

Reply:
https://github.com/harvester/harvester/blob/v0.2/deploy/charts/harvester/Chart.yaml#L56 it is already using 1.1.1, not sure how u get the 1.1.0?

Reply:
Our go.mod still uses v1.1.0. Both v0.2 and master branch

https://github.com/harvester/harvester/blob/v0.2/go.mod#L60
https://github.com/harvester/harvester/blob/158a35d/go.mod#L68

Reply:
If it's applicable, I'll try upgrading Longhorn dependency to v1.1.1

Reply:
PR Updated. PTAL thanks!